### PR TITLE
Ac-5185 - allow impact-api to deploy a release tag

### PR DESCRIPTION
**Pull request by shanbady**
Found the golden tip here which basically suggests asking the remote repo for tag information (since travis does some odd things with the local origin):
https://github.com/travis-ci/travis-ci/issues/4745

To test this 

- make a branch off of this branch (Ac-5185b)
- make a small change and push it
- observe build and see that an image with the name of your branch is tagged in ECR (https://console.aws.amazon.com/ecs/home?region=us-east-1#/repositories/impact-api#images;tagStatus=ALL)

- publish a new test release off of your branch - give it "test tag" of some sort (instead of the normal release tag format ex: 2017.12.12)
- allow travis to run the build and on completion observe that an image with the name of the release tag (not the branch) shows up in ECR (https://console.aws.amazon.com/ecs/home?region=us-east-1#/repositories/impact-api#images;tagStatus=ALL)

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -43,4 +48,10 @@ after_success:
 - pip install awscli;
 - export PATH=$PATH:$HOME/.local/bin;
 - eval $(aws ecr get-login --region us-east-1 --no-include-email);
-- make release AWS_ACCESS_KEY_ID=$AWS_ACCESS_KEY_ID AWS_SECRET_ACCESS_KEY=$AWS_SECRET_ACCESS_KEY ENVIRONMENT=$DEPLOY_ENVIRONMENT DOCKER_REGISTRY=$DOCKER_REGISTRY RELEASE_TAG=$BRANCH
+- git remote rm origin
+- git remote add origin https://github.com/masschallenge/impact-api.git 
+- git fetch
+- echo $REAL_BRANCH
```
Comment (yotammanor) [inline .travis.yml]: Is this debugging leftover or is it being used?

File: .travis.yml [hunk 2]
```diff
@@ -43,4 +48,10 @@ after_success:
...
+- export BRANCH=$(if [ -z $TRAVIS_TAG]; then echo $(git describe --exact-match $TRAVIS_TAG); else echo $TRAVIS_BRANCH; fi)
+- echo $BRANCH
```
Comment (yotammanor) [inline .travis.yml]: likewise

**Review by yotammanor — CHANGES_REQUESTED** [review]
Functional error:
- See [This build#L1209](https://travis-ci.org/masschallenge/impact-api/builds/324827928#L1209) - the value of BRANCH ends up being null - and of course the branch name does not appear in the ECR.

**Review by pbx — CHANGES_REQUESTED** [review]
I worked my way through the instructions. On the final point there, while [my build](https://travis-ci.org/masschallenge/impact-api/builds/325104278) _did_ identify my tag (`Tag test-AC-5185-discard`) up top, no image with that tag is in the ECR list that I can see.

**Comment by pbx** [comment]
After a long troubleshooting walkthrough with Shankar I gave it another try. Unfortunately, same failure.
* My release was tagged (and titled) [test-AC-5185](https://github.com/masschallenge/impact-api/releases/tag/test-AC-5185)
* The [build](https://travis-ci.org/masschallenge/impact-api/builds/325183036) of that tag succeeded
* But the [release tag](url) `test-5185-2` is **not in the ECR image list**

**Comment by shanbady** [comment]
@pbx @yotammanor this is ready to be tested once again. I was able to work around the multi-commit tagging issue last night.

File: .travis.yml [hunk 3]
```diff
@@ -6,7 +6,11 @@ services:
   - docker
 
 env:
-    DOCKER_COMPOSE_VERSION: 1.10.1
+    global:
+        - DOCKER_COMPOSE_VERSION=1.10.1
+        # get all the branches referencing this commit
+        - IS_RELEASE=$(git ls-remote origin | grep "$TRAVIS_COMMIT\s\+refs/heads/release$"
```
Comment (yotammanor) [inline .travis.yml]: missing closing parenthesis

File: .travis.yml [hunk 4]
```diff
@@ -6,7 +6,11 @@ services:
...
+        - RELEASE_TAG=$(git ls-remote  --tags | awk '{print $2, $1}'  | awk -F"/" '{print $3 }' | sort -n -t. -k1,1 -k2,2 -k3,3 | grep -i $TRAVIS_COMMIT | cut -d\  -f 1)
```
Comment (yotammanor) [inline .travis.yml]: Question:
When I'm running locally:
```
MC-HQ-YMANOR:impact-api yotammanor$ export TRAVIS_COMMIT=51ba1c421db7d7d3a5bd411b91b64b910ec365bd
MC-HQ-YMANOR:impact-api yotammanor$ git ls-remote  --tags | awk '{print $2, $1}'  | awk -F"/" '{print $3 }' | sort -n -t. -k1,1 -k2,2 -k3,3 | grep -i $TRAVIS_COMMIT | cut -d\  -f 1
From https://github.com/masschallenge/impact-api.git
test-AC-5185-review-1
test-yotam
test-yotam-2017.1.4
```
Is this the expected output? (specifically - multi line, and the first line is **From ...git**)

Doing:
```
MC-HQ-YMANOR:impact-api yotammanor$ FOO=$(git ls-remote  --tags | awk '{print $2, $1}'  | awk -F"/" '{print $3 }' | sort -n -t. -k1,1 -k2,2 -k3,3 | grep -i $TRAVIS_COMMIT | cut -d\  -f 1)
MC-HQ-YMANOR:impact-api yotammanor$ echo $FOO
test-AC-5185-review-1 test-yotam test-yotam-2017.1.4
```

looks better (although how the omission of the first line happens is not clear to me). Is that correct?

**Review by yotammanor — CHANGES_REQUESTED** [review]
Functionally looks good, handles branch tagging and  two release on one commit tagging as well.

Blocking because of the missing parenthesis, plus a short question which is probably not blocking.

File: .travis.yml (hunk 4 above)
Comment (shanbady) [reply .travis.yml]: I came across this locally. If you try and pipe the output to a file or anywhere else it seems that the first line of output is not actual command output and is ignored so enclosing this in $() has the same effect. it shouldn't cause issues in the variable that gets set in travis.

**Review by yotammanor — APPROVED** [review]

**Review by pbx — CHANGES_REQUESTED** [review]
Similar results as before. I followed the steps as written. An image with the name of my branch *does* show up in ECR. However, though my [build](https://travis-ci.org/masschallenge/impact-api/builds/328206697) succeeded, and identified the tag of [my release](https://github.com/masschallenge/impact-api/releases/tag/test-AC-5185-review-3), the tag (`test-AC-5185-review-3` does not show up in the ECR image list that I can see.

The target of my release was `master`, in case that's salient (it's not specified in the instructions and isn't displayed on the release's page).

**Comment by yotammanor** [comment]
@pbx it needs to be a branch that is downstream from this current PR's commit, or it wouldn't work.

**Review by pbx — APPROVED** [review]

**Pull request merged**